presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetHiveRecordCursor.java

```diff
@@ -722,18 +739,18 @@ private static GroupedConverter createGroupConverter(Type prestoType, String col
         private static final int NULL_BUILDER_SIZE_IN_BYTES_THRESHOLD = 32768;
 
         private final Type rowType;
-        private final int fieldIndex;
+        private int fieldIndex;
 
-        private final List<BlockConverter> converters;
+        private final List<BlockFieldConverter> converters;
         private BlockBuilder builder;
         private BlockBuilder nullBuilder; // used internally when builder is set to null
         private BlockBuilder currentEntryBuilder;
 
-        public ParquetStructConverter(Type prestoType, String columnName, GroupType entryType, int fieldIndex)
+        private int[] converterFieldIndex;
+        private InterleavedBlockBuilder tmpBuilder; // a tmp builder for fields out of order
```

Comment:
let's not call it tmpBuilder, call it outofOrderBlockBuilder or parquetSchemaBlockBuilder, what do u think?